### PR TITLE
Enable gcc-4.8 compilation

### PR DESCRIPTION
**Pull request by loganek**
Once this PR is merged, I'll think of having a CI job for gcc4.8 too

### CODE REVIEW

**Comment by loganek** [comment]
> why do you care that old gcc? (just a curious question)

Unfortunately some of the toolchains still use that version.

File: core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/numeric_limits.h [hunk 1]
```diff
@@ -39,4 +53,21 @@
              : ULLONG_MAX, default           \
              : (void)0)
 
+#else
```
Comment (loganek) [inline core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/numeric_limits.h:56]: I actually wonder if there's any advantage of keeping the implementation with `_Generic` at all.. any opinions if I remove it?

Comment (wenyongh) [reply core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/numeric_limits.h:56]: There is only one place using NUMERIC_MAX, and NUMBERIC_MIN isn't used:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c#L2260-L2261
```C
    out->tv_sec =
        (time_t)in < NUMERIC_MAX(time_t) ? (time_t)in : NUMERIC_MAX(time_t);
```
I remember that time_t is defined as `long int`, we should be able to simplify the code.

Comment (yamt) [reply core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/numeric_limits.h:56]: please don't assume long int. we have BH_TIME_T_MAX for this purpose.

**Review by wenyongh — COMMENTED** [review]
LGTM

**Pull request merged**